src/Surfnet/ServiceProviderDashboard/Application/CommandHandler/Entity/PublishEntityProductionCommandHandler.php

```diff
@@ -103,32 +133,49 @@ public function handle(PublishEntityProductionCommand $command)
     {
         $entity = $this->repository->findById($command->getId());
 
+        // 1. Create the Jira ticket
+        $ticket = Ticket::fromEntity(
+            $entity,
+            $command->getApplicant(),
+            $this->issueType,
+            $this->summaryTranslationKey,
+            $this->descriptionTranslationKey
+        );
+
+        $this->logger->info(
+            sprintf('Creating a %s Jira issue for "%s".', $this->issueType, $entity->getNameEn())
+        );
+
+        try {
+            $issue = $this->ticketService->createIssueFrom($ticket);
+            $this->logger->info(sprintf('Created Jira issue with key: %s', $issue->key));
+        } catch (Exception $e) {
+            $this->logger->critical('Unable to create the Jira issue.', [$e->getMessage()]);
+
+            // Inform the service desk of the unavailability of Jira
+            $message = $this->mailFactory->buildJiraIssueFailedMessage($e, $entity);
+            $this->mailer->send($message);
+
+            // Customer is presented an error message with the invitation to try again at a later stage
+            $this->flashBag->add('error', 'entity.edit.error.publish');
+            // Stop execution
+            return;
+        }
+
+        // 2. On success, publish the entity to production
         try {
             $this->logger->info(
                 sprintf('Publishing entity "%s" to Manage in production environment', $entity->getNameEn())
             );
-
             $publishResponse = $this->publishClient->publish($entity);
-
             if (array_key_exists('id', $publishResponse)) {
```

Comment:
If manage doesn't return an id, and therefore the entity isn't published it doesn't seems handle this situation properly. And in the end no one would get informed.

Reply:
I don't think Manage will ever return a response without an ID except for error situations, which are handled as Exceptions. The Manage PublishEntityClient only returns an array response if the status of the response was OK. Still it's a small task to add an additional error catch.

Reply:
See 3bfe0e5